sylvia-derive/src/message.rs

```diff
@@ -1333,21 +1352,23 @@ impl<'a> GlueMessage<'a> {
                 #[allow(clippy::derive_partial_eq_without_eq)]
                 #[derive(#sylvia ::serde::Serialize, Clone, Debug, PartialEq, #sylvia ::schemars::JsonSchema)]
                 #[serde(rename_all="snake_case", untagged)]
-                pub enum #enum_name #used_generics {
+                pub enum #contract_enum_name #bracketed_used_generics {
                     #(#variants,)*
                     #contract_variant
                 }
 
-                impl #used_generics #enum_name #used_generics {
-                    pub fn dispatch #unused_generics #where_clause (
+                impl #bracketed_used_generics #contract_enum_name #bracketed_used_generics {
+                    pub fn dispatch #unused_generics (
                         self,
                         contract: &#contract,
                         ctx: #ctx_type,
-                    ) -> #ret_type {
-                        const _: () = {
+                    ) -> #ret_type #full_where_clause {
+                        const fn assert_no_intersection #bracketed_used_generics () #where_clause {
                             let msgs: [&[&str]; #variants_cnt] = [#(#messages_call),*];
                             #sylvia ::utils::assert_no_intersection(msgs);
-                        };
+                        }
+
+                        assert_no_intersection #prefixed_used_generics ();
```

Comment:
But if think it can be compile time. You made a function constant; why not do this:
```suggestion
                        const _: () = assert_no_intersection #prefixed_used_generics ();
```
In my opinion, it should force executing `assert_no_intersection` in the compile time.

Reply:
Unfortunately this causes the problem I originally encountered:

```
use of generic parameter from outer function type parameter from outer function
```

Reply:
Solution I went with is to move the `messages` method out of f.e. `ExecMsg` and have it as a separate function named `execute_messages` or `query_messages` to avoid name collissions.